File: src/main/java/com/github/alexandrecarlton/idea/settings/applier/impl/editor/codestyle/java/JavaImportsSettingsApplier.java
package com.github.alexandrecarlton.idea.settings.applier.impl.editor.codestyle.java;

import com.github.alexandrecarlton.idea.settings.applier.api.SettingsApplier;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.JavaImportsSettings;
import com.intellij.psi.codeStyle.CodeStyleSettings;
import com.intellij.psi.codeStyle.JavaCodeStyleSettings;
import javax.inject.Inject;

public class JavaImportsSettingsApplier implements SettingsApplier<JavaImportsSettings> {

  private final CodeStyleSettings codeStyleSettings;

  @Inject
  public JavaImportsSettingsApplier(CodeStyleSettings codeStyleSettings) {
    this.codeStyleSettings = codeStyleSettings;
  }

  public void apply(JavaImportsSettings imports) {
    final JavaCodeStyleSettings javaCodeStyleSettings = codeStyleSettings.getCustomSettings(JavaCodeStyleSettings.class);
    imports.classCountToUseImportWithWildcard().ifPresent(javaCodeStyleSettings::setClassCountToUseImportOnDemand);
    imports.namesCountToUseStaticImportWithWildcard().ifPresent(javaCodeStyleSettings::setNamesCountToUseImportOnDemand);
  }
}

File: applier/impl/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/editor/codestyle/java/imports/JavaImportsSettingsApplier.kt
package com.github.alexandrecarlton.idea.settings.dialog.editor.codestyle.java.imports

import com.github.alexandrecarlton.idea.settings.dialog.SettingsApplier
import com.intellij.psi.codeStyle.JavaCodeStyleSettings
import javax.inject.Inject

class JavaImportsSettingsApplier @Inject
constructor(private val javaCodeStyleSettings: JavaCodeStyleSettings) : SettingsApplier<JavaImportsSettings> {

    override fun apply(settings: JavaImportsSettings) {
        settings.classCountToUseImportWithWildcard?.let { javaCodeStyleSettings.classCountToUseImportOnDemand = it }
        settings.namesCountToUseStaticImportWithWildcard?.let { javaCodeStyleSettings.namesCountToUseImportOnDemand = it }
    }
}
